core/trino-main/src/main/java/io/trino/operator/output/UnnestingPositionsAppender.java

```diff
@@ -211,6 +211,14 @@ public long getSizeInBytes()
                 (rleValue != null ? rleValue.getSizeInBytes() : 0);
     }
 
+    void addSizesToAccumulator(PositionsAppenderSizeAccumulator accumulator)
+    {
+        long sizeInBytes = getSizeInBytes();
+        // dictionary size is not included due to the expense of the calculation, so this will under-report for dictionaries
```

Comment:
`dictionary size is not included due to the expense of the calculation`
nit: actually there is `io.trino.operator.output.UnnestingPositionsAppender#dictionary` so it's easy to account for it.

Reply:
It’s easy to count for the retained size, but not for which positions are actually in the output and the size of those positions without expensive book keeping.